jupyter_console/ptshell.py

```diff
@@ -708,22 +708,22 @@ def handle_iopub(self, msg_id=''):
     def handle_rich_data(self, data):
         for mime in self.mime_preference:
             if mime in data and mime in self._imagemime:
-                self.handle_image(data, mime)
-                return True
+                return self.handle_image(data, mime)
 
     def handle_image(self, data, mime):
         handler = getattr(
             self, 'handle_image_{0}'.format(self.image_handler), None)
         if handler:
-            handler(data, mime)
+            return handler(data, mime)
 
     def handle_image_PIL(self, data, mime):
         if mime not in ('image/png', 'image/jpeg'):
-            return
+            return False
         import PIL.Image
         raw = base64.decodestring(data[mime].encode('ascii'))
         img = PIL.Image.open(BytesIO(raw))
         img.show()
```

Comment:
This can fail, but won’t say so. `ImageShow.show` returns `1` or `0` if it works/doesn’t